src/main/java/uk/gov/hmcts/reform/ccd/client/CoreCaseDataApi.java

```diff
@@ -201,6 +201,23 @@ CaseDetails submitEventForCitizen(
             @RequestParam Map<String, String> searchCriteria
     );
 
+    /**
+     * Retrieve case by Case Id
+     */
+    @RequestMapping(
```

Comment:
Since this is `V2` I think we should create a new Client for V2 api's
CoreCaseDataApiV2.java
CcdApiV2.java ?
CoreCaseDataApi.V2.java

This way untangle the Models and add the linking functionality to them

Reply:
There shouldn't be versioning at that level imo, we're deprecating the old api in this PR and saying use the new one, later we'll remove the old methods and eventually this class will only have the new apis

Reply:
not sure about that ... its a new api and model so we should think about a whole break.
I.e. reference the deprecation to the new api.
Under the hood *I think* they will be using the same connection pool so it's not a big overhead.

Reply:
Or we could put it on a new interface that this one extends kinda an inheritance/composition ?

Reply:
i think the complexity and pain for everyone are bigger especially then there's not a new api replacement for everything now, 
if it was a clean upgrade sure, but currently you would have to interchange quite a lot...

Reply:
We cant compose since feign does not support inheritance 😢